pkg/build/controller/build/build_controller.go

```diff
@@ -956,12 +996,46 @@ func (bc *BuildController) updateBuild(build *buildapi.Build, update *buildUpdat
 			bc.recorder.Eventf(patchedBuild, kapi.EventTypeNormal, buildapi.BuildFailedEventReason, fmt.Sprintf(buildapi.BuildFailedEventMessage, patchedBuild.Namespace, patchedBuild.Name))
 		}
 		if buildutil.IsTerminalPhase(*update.phase) {
-			common.HandleBuildCompletion(patchedBuild, bc.buildLister, bc.buildConfigGetter, bc.buildDeleter, bc.runPolicies)
+			bc.handleBuildCompletion(patchedBuild)
 		}
 	}
 	return nil
 }
 
+func (bc *BuildController) handleBuildCompletion(build *buildapi.Build) {
+	bcName := buildutil.ConfigNameForBuild(build)
+	bc.enqueueBuildConfig(build.Namespace, bcName)
+	if err := common.HandleBuildPruning(bcName, build.Namespace, bc.buildLister, bc.buildConfigGetter, bc.buildDeleter); err != nil {
+		utilruntime.HandleError(fmt.Errorf("failed to prune old builds %s/%s: %v", build.Namespace, build.Name, err))
+	}
```

Comment:
seems like this could also run inside handleBuildConfig.  Not sure if there's an advantage to moving it though.

Reply:
leaving as is given that in handleBuildConfig it could potentially be called multiple times which could be wasteful.